component/core/src/main/java/org/exoplatform/social/core/listeners/GroupSynchronizationSocialProfileListener.java

```diff
@@ -0,0 +1,164 @@
+  }
+
+  @Override
+  public void avatarUpdated(ProfileLifeCycleEvent event) {
+    profileUpdated(event);
+  }
+
+  @Override
+  public void bannerUpdated(ProfileLifeCycleEvent event) {
+    profileUpdated(event);
+  }
+
+  @Override
+  public void contactSectionUpdated(ProfileLifeCycleEvent event) {
+    profileUpdated(event);
+
+  }
+
+  @Override
+  public void experienceSectionUpdated(ProfileLifeCycleEvent event) {
+    profileUpdated(event);
+
+  }
+
+  @Override
+  public void createProfile(ProfileLifeCycleEvent event) {
+    profileUpdated(event);
+
+  }
+
+  private void profileUpdated(ProfileLifeCycleEvent profileLifeCycleEvent) {
+    Profile profile = profileLifeCycleEvent.getProfile();
+    User user;
+    try {
+      user = organizationService.getUserHandler().findUserByName(profile.getIdentity().getRemoteId());
+      synchronizePropertyGroups(profileLifeCycleEvent.getProfile(), user, profilePropertySettingsService, organizationService);
+    } catch (Exception e) {
+      LOG.error("Error while synchronizing profile property groups");
+    }
+  }
+
+  private void synchronizePropertyGroups(Profile profile,
+                                         User user,
+                                         ProfilePropertySettingsService profilePropertySettingsService,
+                                         OrganizationService organizationService) {
+    Group profileGroup = createGroup(PROFILE_GROUP_NAME, null, organizationService);
+    if (profileGroup != null) {
+      List<String> synchronizedProperties = profilePropertySettingsService.getSynchronizedPropertySettings()
+                                                                          .stream()
+                                                                          .filter(ProfilePropertySetting::isActive)
+                                                                          .map(ProfilePropertySetting::getPropertyName)
+                                                                          .toList();
+      Set<Map.Entry<String, Object>> properties = profile.getProperties().entrySet();
+      properties.stream().filter(property -> synchronizedProperties.contains(property.getKey())).forEach(property -> {
+        String propertyName = property.getKey();
+        String propertyValue = (String) property.getValue();
+        Group newPropertyNameGroup = createGroup(propertyName, profileGroup, organizationService);
+        if (newPropertyNameGroup != null) {
+          Group newPropertyValueGroup = createGroup(propertyValue, newPropertyNameGroup, organizationService);
+          addUserToGroup(newPropertyValueGroup, user, organizationService);
+        }
+      });
+    }
+  }
+
+  private Group getGroup(String groupId, OrganizationService organizationService) {
+    try {
+      GroupHandler groupHandler = organizationService.getGroupHandler();
+      return groupHandler.findGroupById(groupId);
+    } catch (Exception e) {
+      return null;
```

Comment:
Please avoid swallowing an exception and explain the causes and consequences in message with additional information for troubleshooting.

But I think if an error happens here, it should be treated outside this method and handled by upper layer which has to log the error + make the adequate consequences, inside `synchronizePropertyGroups` loop by example, for each item to treat.

```suggestion
      LOG.warn("Error while .... using parameters {} .... This may cause ... Consequently will be handled by ....", param1, param2, param3...., e);
```

Reply:
I find swallowing the exception here is safe as it's already handled in the wrapper function and i don't see a need for throwing. but i agree with that we need to log the error to enhance the situation here.

Reply:
nope, swallowing an exception isn't safe in fact, but the modification is sufficient, since you added the exception in a dedicated log entry.

Reply:
> it's already handled in the wrapper function and i don't see a need for throwing

Please throw an exception to caller, else the business logic of `createGroup` will be incoherent. When an error occurs while accessing database to get an existing group with that name, the behavior inside `createGroup` method must stop. Even more than that, the loop inside `` must stop to not treat the propertyName 
```java
      properties.stream().filter(property -> synchronizedProperties.contains(property.getKey())).forEach(property -> {
        String propertyName = property.getKey();
        String propertyValue = (String) property.getValue();
  try {
  } catch (Exception e) {
    LOG.warn("Error proceeding the synchronization of property {} with value {} for user {}. Ignoring synchronization for this property and continue processing other properties.",propertyName, propertyValue, user.getUserName(), e);
  }
```

Besides, please make the content of the loop in a dedicated method to reduce complexity.

Reply:
not considered yet for `Please throw an exception to caller`